packages/core/src/components/event-lake.aggregate-store.suite.ts

```diff
@@ -0,0 +1,336 @@
+        id: event.payload.accountId,
+        bankId: event.payload.bankId,
+        index: event.payload.index,
+        balance: 0,
+      });
+    }
+
+    deposit(amount: number) {
+      this.apply(
+        Deposited.new({ accountId: this.id, bankId: this.bankId, amount }),
+      );
+    }
+
+    @On(Deposited)
+    onDeposited(event: Deposited) {
+      this.balance += event.payload.amount;
+    }
+
+    withdraw(amount: number) {
+      this.apply(
+        Withdrawn.new({ accountId: this.id, bankId: this.bankId, amount }),
+      );
+    }
+
+    @On(Withdrawn)
+    onWithdrawn(event: Withdrawn) {
+      this.balance -= event.payload.amount;
+    }
+  }
+
+  class Bank extends Derive(
+    Named("Bank"),
+    Shaped({
+      id: BankId,
+      accounts: Number,
+    }),
+  ) {
+    increment(shouldFailInMiddle = false) {
+      this.accounts++;
+      if (this.accounts > 10 && shouldFailInMiddle) {
+        throw new Error("Too many accounts");
+      }
+      return this.accounts;
+    }
+
+    static new() {
+      return new Bank({
+        id: BankId.generate(),
+        accounts: 0,
+      });
+    }
+  }
+
+  const registry = new SerializerRegistry()
+    .auto(AccountOpened)
+    .auto(Deposited)
+    .auto(Withdrawn)
+    .auto(Account)
+    .auto(Bank);
+
+  const bankStore = config.getStore("Bank", AutoSerializer.first(Bank));
+
+  it("save and load", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+    const account = Account.open();
+    await accountStore.save(account);
+    const loadedAccount = await accountStore.load(account.id);
+    expect(loadedAccount).toEqual(account);
+    expect(loadedAccount?.balance).toEqual(0);
+  });
+
+  it("supports concurrent writes, using transactions", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+    const account = Account.open();
+    await accountStore.save(account);
+
+    await Promise.all(
+      [...Array(5).keys()].map(async () => {
+        await config.transaction.perform(async (trx) => {
+          const fresh = await accountStore.load(account.id, trx);
+          fresh!.deposit(1);
+          await accountStore.save(fresh!, trx);
+        });
+      }),
+    );
+
+    const loadedAccount = await accountStore.load(account.id);
+    expect(loadedAccount?.balance).toEqual(5);
+  });
+
+  it("is fast", async () => {
+    const accountStore = config.getAggregateStore(Account, registry);
+
+    const accounts = [...Array(200).keys()].map(() => Account.open());
+
+    const expectMS = async (ms: number, fn: () => Promise<any>) => {
+      const before = process.hrtime.bigint();
+      await fn();
+      const after = process.hrtime.bigint();
+      expect(Number(after - before) / 1_000_000).toBeLessThan(ms);
+    };
+
+    await expectMS(5000, async () => {
+      for (const account of accounts) {
+        await accountStore.save(account);
+      }
+    });
+
+    await expectMS(5000, async () => {
```

Comment:
can we add a 'is fast' test with transactions ?

Reply:
I dont experience the need to for now, but maybe with a future implementation it will be needed